test/caching-precompiler.js

```diff
@@ -48,3 +52,64 @@ test('adds files and source maps to the cache directory as needed', function (t)
 	t.is(files.filter(endsWithMap).length, 1, 'one .map file is saved to the cache');
 	t.end();
 });
+
+test('uses default babel options when babelConfig === "default"', function (t) {
+	var tempDir = uniqueTempDir();
+	var precompiler = new CachingPrecompiler(tempDir, 'default');
+	babel.transform.reset();
+
+	precompiler.precompileFile(fixture('es2015.js'));
+
+	t.true(babel.transform.calledOnce);
+	var options = babel.transform.firstCall.args[1];
+
+	t.true('filename' in options);
+	t.true(options.sourceMaps);
+	t.false(options.ast);
+	t.true('inputSourceMap' in options);
+	t.false(options.babelrc);
+	t.true(Array.isArray(options.presets));
+	t.true(Array.isArray(options.plugins));
+	t.end();
+});
+
+test('allows babel config from package.json/babel when babelConfig === "inherit"', function (t) {
+	var tempDir = uniqueTempDir();
+	var precompiler = new CachingPrecompiler(tempDir, 'inherit');
+	babel.transform.reset();
+
+	precompiler.precompileFile(fixture('es2015.js'));
+
+	t.true(babel.transform.calledOnce);
+	var options = babel.transform.firstCall.args[1];
+
+	t.true('filename' in options);
+	t.true(options.sourceMaps);
+	t.false(options.ast);
+	t.true('inputSourceMap' in options);
+	t.true(options.babelrc);
+	t.end();
+});
+
+test('uses babelConfig for babel options when babelConfig is an object', function (t) {
+	var tempDir = uniqueTempDir();
+	var precompiler = new CachingPrecompiler(tempDir, {
+		presets: ['stage-2', 'es2015'],
+		plugins: []
+	});
+	babel.transform.reset();
+
+	precompiler.precompileFile(fixture('es2015.js'));
+
+	t.true(babel.transform.calledOnce);
+	var options = babel.transform.firstCall.args[1];
+
+	t.true('filename' in options);
+	t.true(options.sourceMaps);
+	t.false(options.ast);
+	t.true('inputSourceMap' in options);
+	t.false(options.babelrc);
+	t.deepEqual(options.presets, ['stage-2', 'es2015']);
+	t.equal(options.plugins.length, 2);
```

Comment:
I can certainly add a fake plugin, which would increase the expected plugins.length to 3. I can also verify that transformRuntime is added because I can just require() it in to get a reference to it and use === for comparison.

But I'm not sure how I can verify that powerAssert was added because it is instantiated inside of the `_factory` function, so I don't have a reference to it. That's why I only verified that lenth === 2.

Is there another way I can validate that the powerAssert plugin was added?